packages/connectivity/src/scp-cf/destination/get-subscriber-token.ts

```diff
@@ -72,34 +74,40 @@ async function retrieveUserToken(
 
 async function retrieveServiceToken(
   options: DestinationOptions,
-  isXsuaaJwt: boolean
+  decodedUserJwt: JwtPayload | undefined
 ): Promise<JwtPair | undefined> {
-  const jwt = getJwtForServiceToken(options, isXsuaaJwt);
+  const jwt = getJwtForServiceToken(options?.iss, decodedUserJwt);
 
   if (jwt) {
-    return getJwtPair(
-      await serviceToken('destination', {
-        ...options,
-        jwt
-      })
-    );
+    try {
+      return getJwtPair(
+        await serviceToken('destination', {
+          ...options,
+          jwt
+        })
+      );
+    } catch (err) {
+      logger.warn(
+        'Failed to fetch subscriber service token for destination. This is only relevant if you are using subscriber destinations.'
```

Comment:
> currently this discards the error, maybe print it to debug to give users some options to get to the root cause, in case they need to? At least that's what we typically do in Java..

I think it is a nice idea to add the error. Debug would hide it even more. I could just add it to the warning. What do you think?


> Also, I'm wondering what the reason for this change is, considering that this threw before. What changed?

This is not obvious from the code at first glance, but previously this request was only executed (and thereby potentially throwing) when there was an `iss` property set or the given token was an XSUAA token. Now it would be executed for all tokens, if given. 
There might be cases where I don't care about subscriber destinations, therefore I wouldn't need a subscriber token and wouldn't care about it failing.
I considered rethrowing the error in case the `iss` property was set or the token was an XSUAA token instead. Do you think that would be better?

Reply:
I still don't understand this "subscriber" aspect, so I don't know what this call is supposed to do, and in which cases this call failing can have a reasonable recovery..

Reply:
This function is called to fetch a token for the subscriber. As we currently ALWAYS retrieve a subscriber token, when a user JWT is available, it is possible that we call this, although it isn't needed, because users are actually in the provider case. This is a very obvious flaw, but I don't dare touching this at the moment.